src/checkers/inference/util/CopyUtil.java

```diff
@@ -154,12 +158,38 @@ private static void copyAnnotationsImpl(final AnnotatedTypeMirror from, final An
         } else if (fromKind == NONE || fromKind == NULL || fromKind == VOID ||
                 toKind == NONE || toKind == NULL || fromKind == NULL) {
              // No annotations
+        } else if (fromKind == INTERSECTION && toKind == INTERSECTION) {
+            AnnotatedIntersectionType fromIntersec = (AnnotatedIntersectionType) from;
+            AnnotatedIntersectionType toIntersec = (AnnotatedIntersectionType) to;
+            List<AnnotatedDeclaredType> fromSuperTypes = fromIntersec.directSuperTypes();
+            List<AnnotatedDeclaredType> toSuperTypes = toIntersec.directSuperTypes();
+            Map<DeclaredType, AnnotatedDeclaredType> fromSuperTypesMap = new HashMap<>();
+            Map<DeclaredType, AnnotatedDeclaredType> toSuperTypesMap = new HashMap<>();
+            for (AnnotatedDeclaredType superType : fromSuperTypes) {
+                fromSuperTypesMap.put(superType.getUnderlyingType(), superType);
+            }
+            for (AnnotatedDeclaredType superType : toSuperTypes) {
+                toSuperTypesMap.put(superType.getUnderlyingType(), superType);
+            }
+
+            //quick check size first
+            if (fromSuperTypesMap.size() != toSuperTypesMap.size()) {
```

Comment:
Move this simple check to before creating either of the Maps and check the lengths of fromSuperTypes and toSuperTypes. No point in constructing the Maps if their lengths won't match.